src/pilot/deploy-overcloud.py

```diff
@@ -30,18 +37,370 @@
+            except Exception as error:
+                message = "Exception {}: {}".format(type(error).__name__,
+                                                    str(error))
+                logger.error("{}".format(message))
+
+            # Create aggregate for hugepages configuration
+            try:
+                if enable_hugepages:
+                    aggregate_name = "HugePage_Aggr"
+                    aggregate_meta = [{'hugepages': 'True'}]
+                    config_oc_obj.configure_dell_aggregate(aggregate_name,
+                                                           aggregate_meta,
+                                                           host_name_list)
+            except Exception as error:
+                message = "Exception {}: {}".format(type(error).__name__,
+                                                    str(error))
+                logger.error("{}".format(message))
+
+            # Create normal aggregate
+            aggregate_name = "Normal_Aggr"
+            aggregate_meta = [{'pinned': 'False'}, {'hugepages': 'False'}]
+            config_oc_obj.configure_dell_aggregate(aggregate_name,
+                                                   aggregate_meta)
+    except Exception as error:
+        message = "Exception {}: {}".format(type(error).__name__, str(error))
+        logger.error("{}".format(message))
+
+
+def get_dell_compute_nodes_uuids():
+    logger.info("Getting dell compute node uuids")
+    # Get the undercloud details
+    os_auth_url, os_tenant_name, os_username, os_password = \
+        CredentialHelper.get_undercloud_creds()
+
+    kwargs = {'username': os_username,
+              'password': os_password,
+              'auth_url': os_auth_url,
+              'project_id': os_tenant_name}
+
+    # Create nova client object
+    nova = nova_client.Client(2, **kwargs)
+
+    # Create the ConfigOvercloud object
+    config_oc_obj = ConfigOvercloud(nova)
+    dell_compute_nodes_uuid = config_oc_obj.get_dell_compute_nodes_uuid()
+    logger.debug("dell_compute_nodes_uuid: {}".format(dell_compute_nodes_uuid))
+    return dell_compute_nodes_uuid
+
+
+def reboot_compute_nodes():
+    logger.info("Rebooting dell compute nodes")
+    # Get the undercloud details
+    os_auth_url, os_tenant_name, os_username, os_password = \
+        CredentialHelper.get_undercloud_creds()
+
+    kwargs = {'username': os_username,
+              'password': os_password,
+              'auth_url': os_auth_url,
+              'project_id': os_tenant_name}
+
+    n_client = nova_client.Client(2, **kwargs)
+
+    dell_compute = []
+    for compute in n_client.servers.list(detailed=False,
+                                         search_opts={'name': 'compute'}):
+        dell_compute.append(n_client.servers.ips(compute)['ctlplane'][0]['addr'])
+
+        logger.info("Rebooting server with id: ".format(compute_id))
+        n_client.servers.reboot(compute.id)
+
+        retries = 32
```

Comment:
The way the code is written now, i t will reboot one server and then execute the following code to wait.  We want it to reboot all the servers and then execute the wait code.  You just need to unindent lines 370-394 by 4 spaces to make it work this way.

Reply:
Well done! Fixed.

Reply:
Fixed.